presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -819,6 +797,7 @@ protected Type visitSubqueryExpression(SubqueryExpression node, AnalysisContext
             Type type = Iterables.getOnlyElement(descriptor.getVisibleFields()).getType();
 
             expressionTypes.put(node, type);
+            scalarSubqueries.add(node);
```

Comment:
this are all subqueries - not only scalar.
This method throws SemanticException if it spots subquery which returns multiple columns. But that does not ensure that subquery is scalar as it still may return multiple rows.